src/sidebar/components/hooks/use-user-filter-options.js

```diff
@@ -0,0 +1,79 @@
+import { useMemo } from 'preact/hooks';
+
+import { useStoreProxy } from '../../store/use-store';
+import { username } from '../../util/account-id';
+
+/** @typedef {import('../../store/modules/filters').FilterOption} FilterOption */
+
+/**
+ * Generate a list of users for filtering annotations; update when set of
+ * annotations or filter state changes meaningfully.
+ *
+ * @return {FilterOption[]}
+ */
+export default function useUserFilterOptions() {
+  const store = useStoreProxy();
+  const annotations = store.allAnnotations();
+  const focusFilters = store.getFocusFilters();
+  const showDisplayNames = store.isFeatureEnabled('client_display_names');
+
+  return useMemo(() => {
+    // Determine unique users (authors) in annotation collection
+    const users = {};
+    annotations.forEach(annotation => {
+      // Display a username, not an account ID. If we're dealing with an
+      // account ID, we need to username() it
+      const _username = username(annotation.user)
```

Comment:
As a stylistic convention we tend to use a trailing rather than leading underscore when there is a need to disambiguate.

Since we're only showing saved annotations in the notebook, the `annotation.user` field should always be populated with a valid value and hence the ternary should be unnecessary here.

Reply:
Addressed.